### PR TITLE
InstallPopover: Rewrite as ListBox

### LINKED ISSUE #3: Pressing enter on search has no effect
To reproduce :
- open "Language & Region" plug
- click on the + in the sidebar (the popover appears)
- search for a language
- press enter

Pressing enter has no effect.

Expected :
Pressing enter should add the highlighted langage

Launchpad Details: [#LP1516125](https://bugs.launchpad.net/bugs/1516125) Lasne Olivier - 2015-11-13 19:00:44 +0000

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/44129287-pressing-enter-on-search-has-no-effect?utm_campaign=plugin&utm_content=tracker%2F60175852&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F60175852&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by danrabbit**
Fixes #3 

Screenshot from 2019-09-13 09 39 52@2x

### CODE REVIEW

**Comment by danrabbit** [comment]
@ryonakano I've changed the design to match the Keyboard Layout popover with a more explicit two steps. Select and then button to activate. This should make it more clear which language will be installed

**Comment by kgrubb** [comment]
Following the steps in the related issue, it looks like pressing enter will attempt to authenticate before password prompting. Typing the password in after the first failed attempt works, though. When the install dialog opens, the search goes away as expected.

ezgif com-optimize

Off topic, but something to note (happens in the master branch too, so its not related to this branch specifically), the newly added language doesn't show up in the sidebar until the settings are reopened.

**Comment by ryonakano** [comment]
@danrabbit Looks nice! Although the popover is still not closed until authentication, I guess that's OK. However, as @kgrubb mentions pressing enter in the search entry will attempt to authenticate before password prompting. I can sometimes confirm this issue with the latest master branch too but it seems we face this issue with this branch whenever pressing enter in the search entry, which I guess we might want to fix.

File: src/Widgets/InstallPopover.vala [hunk 1]
```diff
@@ -14,100 +14,149 @@
...
+        load_languagelist ();
+
+        search_entry.grab_focus ();
+
+        list_box.row_selected.connect ((row) => {
+            var langrow = (LangRow) list_box.get_selected_row ();
+
+            button_add.sensitive = row != null;
+            button_add.label = _("Install %s").printf (Utils.translate (langrow.lang, "C"));
+
```
Comment (ryonakano) [inline src/Widgets/InstallPopover.vala]: BTW I guess we don't want this new line.

**Comment by danrabbit** [comment]
I'm thinking this is an issue in the polkit dialog because it happens with other apps too, but I'm really not sure why this keypress gets passed into the dialog

**Comment by kgrubb** [comment]
In that case, I'd think that the polkit trigger is out of the scope of this pr.

To me, it looks like the changes in this pr work as intended, and we can handle the weird polkit issue in another issue/pr?

**Pull request merged**